<launch>
  <!-- 定義外部輸入的參數 -->
  <arg name="size" default="17x12"/>
  <arg name="square" default="0.01"/>
  <arg name="image_topic" default="/usb_cam/image_raw"/>
  <arg name="camera_topic" default="/usb_cam"/>
  <arg name="destination_path" default="$(find calibration)/script/cal_file/cam_cal"/>
  <arg name="image_width" default="960"/>
  <arg name="image_height" default="540"/>

  <!-- 啟動USB攝像頭 -->
  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node">
    <param name="image_width" value="$(arg image_width)"/>
    <param name="image_height" value="$(arg image_height)"/>
  </node>

  <!-- 執行camera_calibration -->
  <node name="camera_calibration" pkg="camera_calibration" type="cameracalibrator.py"
        args="--size $(arg size) --square $(arg square) --no-service-check image:=$(arg image_topic) camera:=$(arg camera_topic)">
  </node>
</launch>